### PR TITLE
Tests for all components

### PR DESCRIPTION
**Pull request by NikitaBondarevv**

### CODE REVIEW

File: src/components/header/__tests__/index.test.tsx [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+    const initialState = { user: { isAuthenticated: true } }
+    const store = mockStore(initialState)
+    const { container } = render(
+      <BrowserRouter>
+        <Provider store={store}>
+          <Header />
+        </Provider>
+      </BrowserRouter>
+    )
+    const authorizedUserElement = container.querySelector(`.${styles.menu}`)
```
Comment (iliatsymbal) [inline src/components/header/__tests__/index.test.tsx:29]: wrong selector

File: src/components/tabs/__tests__/__snapshots__/navigation.test.tsx.snap [hunk 2]
```diff
@@ -15,7 +15,7 @@ exports[`<Navigation /> should match snapshot 1`] = `
           <time
             class="dates"
           >
-            20 March
+            27 March
```
Comment (iliatsymbal) [inline src/components/tabs/__tests__/__snapshots__/navigation.test.tsx.snap:18]: you shouldn't every day update snapshot

File: src/components/userForm/__tests__/__snapshots__/index.test.tsx.snap [hunk 3]
```diff
@@ -44,6 +47,7 @@ exports[`<UserForm /> should match snapshot 1`] = `
       >
         <input
           class="success"
+          data-test="password"
```
Comment (iliatsymbal) [inline src/components/userForm/__tests__/__snapshots__/index.test.tsx.snap:50]: no need

File: jest.config.js [hunk 4]
```diff
@@ -14,5 +14,14 @@ module.exports = {
...
-  testEnvironment: 'jsdom'
+  testEnvironment: 'jsdom',
+
+  coverageThreshold: {
+    'global': {
+      'branches': 90,
+      'functions': 90,
+      'lines': 90,
+      'statements': 90,
+    },
```
Comment (iliatcymbal) [inline jest.config.js:25]: no need quotes

File: src/components/header/__tests__/index.test.tsx [hunk 5]
```diff
@@ -1,10 +1,27 @@
-import { customRender } from '__mocks__/customRender'
 import { Header } from '..'
+import { customRender } from '__mocks__/customRender'
+import { login } from 'store/user'
+import { store } from 'store'
+import { IUser } from 'interfaces/IUser'
+
+jest.mock('../authorizedUser', () => ({
+  AuthorizedUser: () => <div data-test-id="test" />
```
Comment (iliatcymbal) [inline src/components/header/__tests__/index.test.tsx:8]: why you don't use just `id="test"`

File: src/components/header/__tests__/index.test.tsx [hunk 6]
```diff
@@ -1,10 +1,27 @@
...
 
     expect(asFragment()).toMatchSnapshot()
   })
+
+  test('should render <AuthorizedUser /> if "isAuthenticated" true', () => {
+    const firstName = 'test name'
+    store.dispatch(login({ firstName } as IUser))
+
+    const { container } = customRender(<Header />)
+    const authorizedUserElement = container.querySelector(`[data-test-id=test]`)
```
Comment (iliatcymbal) [inline src/components/header/__tests__/index.test.tsx:23]: why string template quotes?

File: src/components/tabs/__tests__/index.test.tsx [hunk 7]
```diff
@@ -1,11 +1,24 @@
...
+    <Tabs loading={false}>
+      <div>test</div>
+    </Tabs>
+    )
 
     expect(asFragment()).toMatchSnapshot()
   })
+
+  test('should render <Preloader /> if loading = true', () => {
+    const { container } = render(<Tabs children={[]} loading={true} />)
```
Comment (iliatcymbal) [inline src/components/tabs/__tests__/index.test.tsx:18]: `={true}` no need

File: src/components/tabs/__tests__/navigation.test.tsx [hunk 8]
```diff
@@ -1,6 +1,10 @@
 import { fireEvent, render, screen } from '@testing-library/react'
 import { Navigation } from '../navigation'
 
+jest.mock('../../../helpers/constans', () => ({
+  getDates: () => 'test date'
```
Comment (iliatcymbal) [inline src/components/tabs/__tests__/navigation.test.tsx:5]: it should be the dates, but not a string

File: src/contracts/__tests__/tasks.test.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}))
+
+const testUrl = `tasks/${task.id}`
+
+describe('getTaskById', () => {
+  test('should return data on request', () => {
+    getTaskById(task.id)
+    expect(request.get).toHaveBeenCalledWith(testUrl)
+  })
```
Comment (iliatcymbal) [inline src/contracts/__tests__/tasks.test.ts:20]: description and expectation are not matched

File: src/contracts/__tests__/tasks.test.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('updateTask', () => {
+  test('should update data on request "updateTask"', () => {
+    updateTask(task)
+    expect(request.put).toHaveBeenCalledWith(testUrl, task)
+  })
+})
+
+describe('deleteTask', () => {
+  test('should work delete on request', () => {
```
Comment (iliatcymbal) [inline src/contracts/__tests__/tasks.test.ts:31]: I can't get from description what does this test do

File: src/contracts/__tests__/tasks.test.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('getTaskById', () => {
+  test('should return data on request', () => {
+    getTaskById(task.id)
+    expect(request.get).toHaveBeenCalledWith(testUrl)
+  })
+})
+
+describe('updateTask', () => {
+  test('should update data on request "updateTask"', () => {
```
Comment (iliatcymbal) [inline src/contracts/__tests__/tasks.test.ts:24]:
```suggestion
  test('should call "request.put()" on "updateTask"', () => {
```
describe what you're doing in the test

File: src/pages/loginForm/__tests__/index.test.tsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    const error = { error: 'test-error' }
+
+    loginMock.mockReturnValueOnce(error)
+
+    const { container } = render(<LoginForm setUser={setUser} />)
+    const form = container.querySelector(`.${styles.loginForm}`)
+
+    await act(() => fireEvent.submit(form!))
+
+    expect(setUser).not.toHaveBeenCalled()
```
Comment (iliatcymbal) [inline src/pages/loginForm/__tests__/index.test.tsx:52]: where do you check error window is shown?

File: src/pages/logout/__tests__/index.test.tsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+    expect(asFragment()).toMatchSnapshot()
+  })
+
+  test('should call logout if don`t error', () => {
+    customRender(<Logout />)
+
+    expect(logout).toHaveBeenCalled()
+  })
+
+  test('should show error text if "logout" don`t work', async () => {
```
Comment (iliatcymbal) [inline src/pages/logout/__tests__/index.test.tsx:26]: doesn't work

File: src/pages/logout/__tests__/index.test.tsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    const mockError = 'something went wrong, please try leater!'
```
Comment (iliatcymbal) [inline src/pages/logout/__tests__/index.test.tsx:27]: there is an error in text

File: src/components/tabs/__tests__/navigation.test.tsx [hunk 15]
```diff
@@ -1,6 +1,10 @@
 import { fireEvent, render, screen } from '@testing-library/react'
 import { Navigation } from '../navigation'
 
+jest.mock('../../../helpers/constans', () => ({
```
Comment (iliatcymbal) [inline src/components/tabs/__tests__/navigation.test.tsx:4]: `../../../` no need

**Pull request merged**